keel-core/src/main/kotlin/com/netflix/spinnaker/keel/services/AdminService.kt

```diff
@@ -32,4 +34,27 @@ class AdminService(
     log.info("Triggering a recheck for $resourceId by clearing our record of the diff")
     diffFingerprintRepository.clear(resourceId)
   }
+
+  /**
+   * Removes the stored state we have for any stateful constraints for an environment
+   * so they will evaluate again
+   */
+  fun redoConstraints(application: String, environment: String, type: String? = null) {
```

Comment:
Can you rename it to `retry` or `reevaluate` instead of `redo`?

Reply:
I can do reevaluate. How come?

Reply:
I think `reevaluate` might reflect better the purpose of this function?  I don't know 🤷

Reply:
seems better to me also

Reply:
Sorry to nitpick, but we're not actually re-evaluating anything here. I think a better name would be `forceConstraintReevaluation`.